app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala

```diff
@@ -25,7 +25,14 @@ final case class MessageDetails(
                                  username: String,
                                  iconEmoji: Option[String] = None,
                                  attachments: Seq[Attachment] = Nil
-                               )
+                               ) {
+
+  def getFields: Array[String] = getClass.getDeclaredFields.map(field => {
```

Comment:
Is reflection necessary here? There are only 4 fields that seem to be known at compile time

Reply:
I've simplified this now